olga/analytics/tests/test_models.py

```diff
@@ -0,0 +1,309 @@
+"""
+Tests for analytics models.
+"""
+
+from datetime import date, datetime
+
+from ddt import ddt, data, unpack
+from mock import patch
+
+from django.test import TestCase
+
+from olga.analytics.tests.factories import EdxInstallationFactory, InstallationStatisticsFactory
+from olga.analytics.models import (
+    EdxInstallation, InstallationStatistics, get_previous_day_start_and_end_dates
+)
+
+# pylint: disable=invalid-name, attribute-defined-outside-init
+
+
+class TestEdxInstallationMethods(TestCase):
+    """
+    Tests for EdxInstallation model's core methods, that work with database.
+    """
+
+    @staticmethod
+    def setUp():
+        """
+        Create one EdxInstallation object.
+        """
+        EdxInstallationFactory(platform_url=None)
+
+    @staticmethod
+    def fake_installation_enthusiast_level():
+        """
+        Create fake edx installation sends statistics with enthusiast level.
+
+        Actually it sends a lot of data, but we need to mock only `platform_url`.
+        """
+        edx_installation_object = EdxInstallation.objects.first()
+        edx_installation_object.platform_url = 'https://url.exists'
+        edx_installation_object.save()
+
+    def test_extending_level_first_time(self):
+        """
+        Verify that does_edx_installation_extend_level_first_time method return True, if not platform name.
+
+        If edx installation has not platform name It means installation sends statistics with `paranoid` level.
+        """
+        edx_installation_object = EdxInstallation.objects.first()
+        result = edx_installation_object.does_edx_installation_extend_level_first_time()
+        self.assertEqual(True, result)
+
+    def test_extending_level_not_first_time(self):
+        """
+        Verify that does_edx_installation_extend_level_first_time method return False, if platform url exists.
+
+        If edx installation has platform name It means installation sends statistics with `enthusiast` level.
+        """
+        self.fake_installation_enthusiast_level()
+
+        edx_installation_object = EdxInstallation.objects.first()
+        result = edx_installation_object.does_edx_installation_extend_level_first_time()
+        self.assertEqual(False, result)
+
+    def test_update_edx_instance_info(self):
+        """
+        Verify that update_edx_instance_info method save edx installation enthusiast data.
+        """
+        enthusiast_edx_installation = {
+            'latitude': 50.32,
+            'longitude': 45.11,
+            'platform_name': 'platform_name',
+            'platform_url': 'https://platform.url'
+        }
+
+        edx_installation_object = EdxInstallation.objects.first()
+        edx_installation_object.update_edx_instance_info(enthusiast_edx_installation)
+
+        extended_edx_installation_object_attributes = edx_installation_object.__dict__
+
+        self.assertDictContainsSubset(enthusiast_edx_installation, extended_edx_installation_object_attributes)
```

Comment:
Closed with https://github.com/raccoongang/acceptor/pull/40/files#r128354553. I think the same for that.